Models/TreeModel.cpp

```diff
@@ -689,13 +689,13 @@ QModelIndex TreeModel::addNode(const Message &child, const QModelIndex &parent)
   return insert(insertParent, pos, child);
 }
 
-QModelIndex TreeModel::duplicateNode(const QModelIndex & /*index*/) {
-  /*Node *node = IndexToNode(index);
-  if (node->repeated_model) {
-    return node->mapFromSource(node->repeated_model->duplicate(mapToSource(index)));
-  }*/
-  qDebug() << rowCount() << " rows";
-  return QModelIndex();
+QModelIndex TreeModel::duplicateNode(const QModelIndex & index) {
+  Node *node = IndexToNode(index);
+  if (node->IsRepeated()) {
+    //return node->mapFromSource(node->repeated_model->duplicate(mapToSource(index)));
+  }
+  Node *parent = IndexToNode(index.parent());
```

Comment:
PLEASE check these; each one of these is capable of returning null, and the safety checks I added only work if you actually verify that I didn't return null.